src/Symfony/Component/HttpFoundation/File/File.php

```diff
@@ -133,4 +133,18 @@ protected function getName($name)
 
         return $originalName;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSize()
+    {
+        // Since stream wrappers and stream filters can change the file contents as read, the only
+        // reliable way to get the effective size of the file is to actually read it.
+        ob_start();
+        $size = readfile($this->getPathname());
+        ob_end_clean();
```

Comment:
You seem to be correct, of course. I think the code worked in the context of my project on account of incidental configuration details, but considering the matter abstractly it obviously doesn't seem like a solution suitable for generic application. Using chunked encoding in case of a wrapper/filter does seem like a sensible approach. I don't know what that would look like, though.